src/Chainweb/Pact/PactService.hs

```diff
@@ -338,9 +338,9 @@ initializeCoinContract _logger v cid pwo = do
 lookupBlockHeader :: BlockHash -> Text -> PactServiceM cas BlockHeader
 lookupBlockHeader bhash ctx = do
   bhdb <- asks _psBlockHeaderDb
-  liftIO $! lookupM bhdb bhash
-        `catch` \e -> throwM $ BlockHeaderLookupFailure $
-                      "failed lookup of parent header in " <> ctx <> ": " <> sshow (e :: SomeException)
+  liftIO $! lookupM bhdb bhash `catchAllSynchronous` \e ->
+        throwM $ BlockHeaderLookupFailure $
+            "failed lookup of parent header in " <> ctx <> ": " <> sshow e
```

Comment:
`catchAllSynchronous` doesn't catch asynchronous exceptions. Catching asynchronous exception interferes with top-level exception handlers. In particular it interferes with thread management from the `async` package.